Sources/ActorSingletonPlugin/ActorSingleton.swift

```diff
@@ -134,3 +52,57 @@ public enum AllocationStrategySettings {
         }
     }
 }
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: ActorSingletonRemoteCallInterceptor
+
+struct ActorSingletonRemoteCallInterceptor<Singleton: DistributedActor>: RemoteCallInterceptor where Singleton.ActorSystem == ClusterSystem {
+    let system: ClusterSystem
+    let proxy: ActorSingletonProxy<Singleton>
+
+    func interceptRemoteCall<Act, Err, Res>(
+        on actor: Act,
+        target: RemoteCallTarget,
+        invocation: inout ClusterSystem.InvocationEncoder,
+        throwing: Err.Type,
+        returning: Res.Type
+    ) async throws -> Res
+        where Act: DistributedActor,
+        Act.ID == ActorID,
+        Err: Error,
+        Res: Codable
+    {
+        // FIXME: better error handling
+        guard actor is Singleton else {
+            fatalError("Wrong interceptor")
+        }
+
+        // FIXME: can't capture inout param
```

Comment:
That's okey/expected I think -- we have to copy / capture by value the invocation... that sounds okey, since we'll store it for later use when we'd need to store and flush later.